lib/ansible/plugins/strategy/__init__.py

```diff
@@ -949,3 +1026,106 @@ def update_active_connections(self, results):
                 if socket_path:
                     if r._host not in self._active_connections:
                         self._active_connections[r._host] = socket_path
+
+
+class NextAction(object):
+    """ The next action after an interpreter's exit. """
+    REDO = 1
+    CONTINUE = 2
+    EXIT = 3
+
+    def __init__(self, result=EXIT):
+        self.result = result
+
+
+class Debugger(cmd.Cmd):
```

Comment:
I wouldn't mind seeing the debugging implementation bits extract to a separate python file (lib/ansible/utils/strategy_debugger.py ?).  That is assuming adding python files to strategy/ that dont have a StrategyModule in them is a problem.